frontend/src/pages/modelServing/screens/projects/utils.ts

```diff
@@ -313,13 +319,21 @@ export const getProjectModelServingPlatform = (
     if (kServeEnabled) {
       return { platform: ServingRuntimePlatform.SINGLE };
     }
-  }
-  if (project.metadata.labels?.[KnownLabels.MODEL_SERVING_PROJECT] === 'true') {
+    if (nimEnabled) {
+      // TODO: this is weird, it relies on KServe today... so it's never "only installed"
+      return { platform: ServingRuntimePlatform.SINGLE };
+    }
+
+    // TODO: unreachable code unless adding a new platform? probably should throw an error
```

Comment:
Is this really unreachable? If the modelmesh label is true (btw, `MODEL_SERVING_PROJECT` is a bad name, the value is `'modelmesh-enabled'`) we'll hit this case right?

Reply:
All 3 if-statements directly above this line are for the 3 "auto select states" -- the last if really is stupid, but it fills out this area better from a readability point of view.

The first if-statement jumps out if there are no serving or more than 1 serving... so that leaves us with "one serving that we don't have checks for" -- aka unreachable code unless there is a problem. I didn't want to put a throw error here in the last moments of 2.16 -- but I think it should have one to help devs know when there is a 4th serving that this is an area that will fall through.

Reply:
This util isn't just for auto-selecting the platform for a new project, it's also for identifying whether a project already has modelmesh or kserve selected. This `else` case is under the `if (project.metadata.labels?.[KnownLabels.MODEL_SERVING_PROJECT] === undefined)` case. The 3 if statements you're referring to are all under that if block, which runs if the project does not have the modelmesh-enabled label (it's a new project in need of auto-select). This else case is for a project that already has modelmesh selected, which correctly returns `ServingRuntimePlatform.MULTI`. The way I read this, it's not unreachable or an error. Am I missing something?

Reply:
Ohh --- your comment isn't about the line below it, it's about the empty space in the if block above it - I misunderstood. Maybe one more newline would help 😅

Reply:
The indentation should have been clear enough, I was just being dumb. This is fine.